addons/timecop/bootstrap.js

```diff
@@ -0,0 +1,40 @@
+const {classes: Cc, interfaces: Ci, utils: Cu} = Components;
+Cu.import("resource://gre/modules/Services.jsm");
+
+const BROWSER_SEARCH_PREF = "browser.search.";
+
+let gInstall = false;
+
+function install(aData, aReason) {
+  // We check for install here instead of in startup because
+  // In Firefox 46, the install reason didn't work properly
+  // in startup in some cases.
```

Comment:
Do you have any details (bug # etc) for this problem? I don't recall running into it before.

Reply:
No, and I only ran into it using about:debugging, but I wanted to be safe. Honestly, in my add-ons I tend to do this anyway because it feels like the install() is more confident than ADDON_INSTALL on startup.